### PR TITLE
Speedup theano generators

### PR DESCRIPTION
**Pull request by oliverlee**
I've rebased this PR: https://github.com/pydy/pydy/pull/70
since it was made over a year ago. 

@moorepants anything that would keep this from being merged?

### CODE REVIEW

**Comment by moorepants** [comment]
Maybe we should only run the tests with theano on the newest released dep versions due to the conflict with NumPy 1.7.

**Comment by moorepants** [comment]
This is the current offering for Theano from the anaconda repos:

```
moorepants@caramelmonkey:~$ conda search theano
Fetching package metadata: ......
theano                       0.5.0                np17py27_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.5.0                np17py26_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.5.0                np16py27_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.5.0                np16py26_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.5.0                np15py27_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.5.0                np15py26_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.5.0                np17py27_1  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.5.0                np17py26_1  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.5.0                np16py27_1  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.5.0                np16py26_1  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.6.0                np19py27_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.6.0                np19py26_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.6.0                np18py27_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.6.0                np18py26_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.6.0                np17py27_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.6.0                np17py26_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.7.0                np19py35_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.7.0                np19py34_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.7.0                np19py33_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                          .  0.7.0                np19py27_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.7.0                np19py26_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.7.0               np110py35_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.7.0               np110py34_0  http://repo.continuum.io/pkgs/free/linux-64/ 
                             0.7.0               np110py27_0  http://repo.continuum.io/pkgs/free/linux-64/ 
```

NumPy 1.9 is the lowest Theano 0.7.0 supports. We could build a theano package for NumPy 1.7 and host it in the anaconda.org pydy channel, only test theano on the newest dep versions, or drop support for NumPy < 1.9.

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ install:
   - source activate test-env
   - if [[ $DEP_VERSIONS == "oldest" ]]; then
       if [[ $TRAVIS_PYTHON_VERSION == 2.7 ]]; then
-        conda install numpy==1.7.0 scipy==0.11.0 sympy==0.7.4.1 cython==0.17 theano==0.6.0;
+        conda install numpy==1.7.0 scipy==0.11.0 sympy==0.7.4.1 cython==0.17 theano==0.7.0;
```
Comment (moorepants) [inline .travis.yml]: I just added a NumPy 1.7 compatible Theano for Linux 64 here:

https://anaconda.org/pydy/theano

You can change this to `conda install -c pydy numpy==1.7.0 scipy==0.11.0 sympy==0.7.4.1 cython==0.17 theano==0.7.0;` to pull packages from the main anaconda repos and our pydy channel. I think that will fix this.

**Comment by moorepants** [comment]
Please update with master.

**Comment by oliverlee** [comment]
NumPy 1.7 and SciPy 0.11 were released Feb 2013 and Sept 2012 respectively. NumPy 1.9 and Scipy 0.14 were released in Sept and May 2014. I suggest we drop support for earlier versions.

**Comment by moorepants** [comment]
Moved version discussion here: https://github.com/pydy/pydy/issues/271

Let's try to merge this as is for now.

**Pull request merged**